2020-02-Spring/1_strings/README.md

```diff
@@ -97,21 +117,70 @@ Go to [Solution](#s3)   [Top](#top)
 <!-- Don't remove -->
 <a name="s1"/>
 
-### 1. SOLUTION 1 TODO :bug:
+### 1. Reverse String
 
-Source: TODO :bug:
+Source: [LeetCode](https://leetcode.com/problems/reverse-string/solution/)
+ 
+#### Optimal Solution
 
-#### Naive/Simple Solution
+<details>
+<summary>Click to see solution</summary>
+ 
+**What does in-place mean?**
 
-TODO :bug:
+By definition, an in-place algorithm is an algorithm which transforms input using no auxiliary data structure.
+ 
+**Two Pointers Approach:**
 
-#### Optimal Solution
+In this approach, two pointers are used to process two array elements at the same time. Usual implementation is to set one pointer in the beginning and one at the end and then to move them until they both meet.
 
-TODO :bug:
+**Algorithm:**
+
+Set pointer left at index 0, and pointer right at index n - 1, where n is a number of elements in the array.
 
+While left < right:
+
+Swap char[left] and char[right].
+
+Move left pointer one step right, and right pointer one step left.
+
+**Java:**
+```
+class Solution {
+    public void reverseString(char[] s) {
+        int left = 0, right = s.length - 1;
+        while (left < right) {
+            char tmp = s[left];
+            s[left++] = s[right];
+            s[right--] = tmp;
+        }
+    }
+}
+
+```
+
+**Python:**
```

Comment:
Feel free to ignore this, but it might be fun to mention that you can make this problem really trivial with Python.

Either with `s.reverse()` or `s[::-1]`

Reply:
I will pretend those methods don't exist.